src/search/pruning/limited_pruning.cc

```diff
@@ -0,0 +1,78 @@
+#include "limited_pruning.h"
+
+#include "../option_parser.h"
+#include "../plugin.h"
+
+#include "../utils/logging.h"
+
+using namespace std;
+
+namespace limited_pruning {
+LimitedPruning::LimitedPruning(const Options &opts)
+    : pruning_method(opts.get<shared_ptr<PruningMethod>>("pruning")),
+      min_required_pruning_ratio(opts.get<double>("min_required_pruning_ratio")),
+      num_expansions_before_checking_pruning_ratio(
+          opts.get<int>("expansions_before_checking_pruning_ratio")),
+      num_pruning_calls(0),
+      is_pruning_disabled(false) {
+}
+void LimitedPruning::initialize(const shared_ptr<AbstractTask> &task) {
+    PruningMethod::initialize(task);
+    pruning_method->initialize(task);
+    utils::g_log << "pruning method: limited" << endl;
+}
+
+void LimitedPruning::prune_operators(
+    const State &state, vector<OperatorID> &op_ids) {
+    if (is_pruning_disabled) {
+        return;
+    }
+    if (min_required_pruning_ratio > 0. &&
+        num_pruning_calls == num_expansions_before_checking_pruning_ratio) {
```

Comment:
Is the intention to only check this once during the entire search, not iteratively after x*num_expansions_before_checking_pruning_ratio?
If so, might it make sense to set min_required_pruning_ratio to 0 inside the if-statement to make the if return false faster in the future? (Genuine question, I don't have an intuition.)

Reply:
Yes, this is indeed the intention. I wouldn't expect it to cause a large overhead. I would not want to change `min_required_pruning_ratio` because that is logically a const variable (and it is actually const). However, to still do something about this: what do you think about ordering `num_pruning_calls == num_expansions_before_checking_pruning_ratio` first? This evaluates to true exactly once.

Reply:
Ah I missed that it was const, good point. I think we can also leave it as is, especially if you planned on not running experiments ;).